src/QirRuntime/test/QIR-static/compiler/QirTarget.qs

```diff
@@ -5,6 +5,26 @@ namespace Microsoft.Quantum.Intrinsic {
 
     open Microsoft.Quantum.Targeting;
 
+    function NAN() : Double {
+        body intrinsic;
+    }
+
+    function IsNan(d: Double) : Bool {
```

Comment:
Are these intended to be new functions in the public Q# API? If so, they will need to go through API review first.

Reply:
(FYI: @bettinaheim, @irinayat-MS, @guenp)
Bettina,
What do I need to do to get those through API review?

I assume that Q#'s `Sqrt()` (inspired by C#'s [Math.Sqrt](https://docs.microsoft.com/en-us/dotnet/api/system.math.sqrt?view=net-5.0)) is expected to work with `NaN` and `PositiveInfinity` also (i.e. the Q#'s `Sqrt()` requires them in Q#). Thus it seems to me reasonable for the functionality of NaN, PositiveInfinity/Infinity, `IsNan()`, `IsInf()` to _become new functions in the public Q# API_ (in simulators can be implemented through the following C++'s <cmath> calls/constants - [`NAN`/`nan()`/`quiet_NaN()`](https://en.cppreference.com/w/cpp/numeric/math/NAN), [`INFINITY`/`infinity`](https://en.cppreference.com/w/cpp/numeric/math/INFINITY), [`isnan()`](https://en.cppreference.com/w/cpp/numeric/math/isnan), [`isinf()`](https://en.cppreference.com/w/cpp/numeric/math/isinf)).

Chris, thanks!

Reply:
Please see the [Q# API Design Process](https://github.com/microsoft/QuantumLibraries/wiki/Q%23-API-Design-Process) documentation on the Q# libraries wiki for more details, but roughly the Q# API Review is a monthly meeting that I run with the Q# API review board (currently @guenp, @msoeken, @bettinaheim, and @efratshabtai) to make sure that all new or modified public Q# APIs meet our [style guide](https://docs.microsoft.com/quantum/resources/contributing/style-guide) and our [API design principles](https://docs.microsoft.com/azure/quantum/contributing-api-design-principles).

Following that process, if we want to add these functions to the public Q# API, my suggestion would be to open a proposal on the [microsoft/quantumlibraries repo](https://github.com/microsoft/QuantumLibraries), so that it can be scheduled for inclusion in the March review. Should the Q# API review approve the new API, and should @bettinaheim sign off on the corresponding impact on the runtime interface, we can then implement those new functions as `body intrinsic` callables in the QSharpFoundation project on the qsharp-runtime repo, with the changes here in QirTarget.qs and bridge-qis.ll being used to implement those new functions.

Reply:
See [413](https://github.com/microsoft/QuantumLibraries/issues/413) in QuantumLibraries.
I propose those for the Q# API Design Review.
@cgranade, @bettinaheim.